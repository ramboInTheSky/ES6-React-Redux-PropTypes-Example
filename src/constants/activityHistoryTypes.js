import { PAUSE } from './pause';
import { GENERIC_PAYMENTPLAN } from './paymentPlan';
import { GENERIC_LEGAL_REFERRAL } from './legalReferral';

// eslint-disable-next-line import/prefer-default-export
export const activityHistoryTypes = {
  PAUSE,
  PAYMENT: 'Payment',
  PAYMENT_PLAN: GENERIC_PAYMENTPLAN,
  LEGAL_REFERRAL: GENERIC_LEGAL_REFERRAL,
  REFERRAL: 'Referral',
  INTERACTION: 'Interaction',
};
