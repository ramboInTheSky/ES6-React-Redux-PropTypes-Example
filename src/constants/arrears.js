export const SYSTEM_PAUSED = 'System Paused';
export const MAUALLY_PAUSED = 'Manually Paused';
