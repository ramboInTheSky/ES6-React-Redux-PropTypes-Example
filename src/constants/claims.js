export const permissions = 'http://nhhg.org.uk/claims/permissions';
export const userType = 'http://nhhg.org.uk/claims/user_type';
