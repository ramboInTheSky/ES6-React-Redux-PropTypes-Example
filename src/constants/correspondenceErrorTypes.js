export const FailedFatal = 'FailedFatal';
export const FailedRetriable = 'FailedRetriable';
