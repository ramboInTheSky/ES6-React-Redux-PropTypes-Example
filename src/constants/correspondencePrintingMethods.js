import path from 'ramda/src/path';
import Dictionary from '../constants/dictionary';

export const LOCAL = 'local';
export const MAILING_HOUSE = 'mailing-house';

const dictionary = Dictionary();

export default [
  {
    id: MAILING_HOUSE,
    name: path(['correspondence', 'labels', 'printViaMailingHouse'], dictionary),
  },
  { id: LOCAL, name: path(['correspondence', 'labels', 'printLocally'], dictionary) },
];
