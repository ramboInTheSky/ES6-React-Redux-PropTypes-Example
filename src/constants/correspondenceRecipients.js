const recipients = Object.freeze([
  { name: 'Customer(s)', value: 'customer' },
  { name: 'Third party', value: 'thirdparty' },
]);

export const CUSTOMER = recipients[0].name;
export const THIRD_PARTY = recipients[1].name;

export default recipients;
