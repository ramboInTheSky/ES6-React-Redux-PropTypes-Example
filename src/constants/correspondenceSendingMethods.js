export const EMAIL = 'Email';
export const LETTER = 'Letter';
export const SMS = 'Sms';
export const SYSTEM = 'System';
