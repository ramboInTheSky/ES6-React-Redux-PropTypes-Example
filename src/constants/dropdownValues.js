export const TASKS_SEL = 'Tasks';
export const ARREARS_SEL = 'Arrears';
export const PRIORITY_SEL = 'priority';
export const CURRENT_BALANCE_SEL = 'currentBalance';
export const DAYS_IN_ARREARS_SEL = 'daysInArrears';
