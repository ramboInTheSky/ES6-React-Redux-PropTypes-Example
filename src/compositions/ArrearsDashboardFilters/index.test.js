import React from 'react';
import { shallow } from 'enzyme';

import { ArrearsFiltersComposition } from './';
import { TASKS_SEL } from '../../constants/dropdownValues';

describe('<ArrearsFiltersComposition />', () => {
  let props;
  let el;

  beforeEach(() => {
    props = {
      arrearsDashboardLabels: {
        customerHeading: 'My arrears',
        housingOfficerHeading: 'Arrears',
        priority: 'Priority',
        currentBalance: 'Current Balance',
        daysInArrears: 'Days in Arrears',
        sortBy: 'Sort By',
        stage: 'Stage',
        show: 'Show',
        selectAStage: 'Select a Stage',
        tasks: 'Tasks',
      },
      filterLabels: {
        allStages: 'All Stages',
        arrears: 'Arrears - {status} ({count})',
        arrearsByStage: '\u00A0\u00A0\u00A0\u00A0{stage} ({count})',
      },
      heading: 'titleText',
      getTasksSummary: jest.fn(),
      getArrearStatuses: jest.fn(),
      getTaskStatuses: jest.fn(),
      arrearStatuses: {
        All: {
          count: 11,
          id: null,
          products: [
            {
              product: 'Leasehold',
              stages: [
                {
                  id: '4cb2fc58-4b75-e811-80c5-00155d072d63',
                  name: 'Stage 2',
                  count: 6,
                },
              ],
            },
          ],
        },
        'In Progress': {
          count: 10,
          id: 1,
          products: [
            {
              product: 'Leasehold',
              stages: [
                {
                  id: '4cb2fc58-4b75-e811-80c5-00155d072d63',
                  name: 'Stage 2',
                  count: 6,
                },
              ],
            },
            {
              product: 'PRH',
              stages: [
                {
                  id: '4cb2fc58-4b75-e811-80c5-00155d072d63',
                  name: '3wks Arrears',
                  count: 2,
                },
                {
                  id: '68027d4f-4b75-e811-80c5-00155d072d63',
                  name: '< 3wks Arrears',
                  count: 2,
                },
              ],
            },
          ],
        },
        'Manually Paused': {
          count: 1,
          id: 857920000,
          products: [
            {
              product: 'Leasehold',
              stages: [
                {
                  id: '4cb2fc58-4b75-e811-80c5-00155d072d63',
                  name: 'Stage 2',
                  count: 1,
                },
              ],
            },
          ],
        },
      },
      taskStatuses: {},
      patches: [
        {
          name: 'John Smith',
          patchName: 'Hello world',
        },
        {
          name: 'Jane Smith',
          patchName: 'Some other name',
        },
      ],
      getArrearsSummary: jest.fn(),
      onOpenPatchSelect: jest.fn(),
      onChange: jest.fn(),
    };

    el = shallow(<ArrearsFiltersComposition {...props} />);
    el.setState({ statusFilterId: null });
  });

  it('should render the page', () => {
    expect(el).toMatchSnapshot();
  });

  it('should call getTaskStatuses', () => {
    expect(props.getTaskStatuses).toHaveBeenCalled();
  });

  it('should call all APIs when changing patches', () => {
    const patches = [{ name: 'rock', patchName: 'roll' }];
    el.setProps({ patches });
    expect(props.getTaskStatuses).toHaveBeenCalled();
    expect(props.getArrearStatuses).toHaveBeenCalled();
    expect(props.getTasksSummary).toHaveBeenCalled();
    expect(props.getArrearsSummary).toHaveBeenCalled();
    expect(props.onChange).toHaveBeenCalledWith(TASKS_SEL);
  });

  it('should call the getArrearsSummary APIs when changing STATUS filters to an Arrears choice', () => {
    const instance = el.instance();
    instance.toggleArrearsOrTasks({ value: 1, label: 'my fantastic label', count: 10 });
    expect(props.getArrearsSummary).toHaveBeenCalledWith(
      { sort: 'priority', status: 1 },
      { value: 1, label: 'my fantastic label', count: 10 }
    );
  });

  it('should call the getTasksSummary APIs when changing STATUS filters to a Tasks choice', () => {
    el.instance().toggleArrearsOrTasks({ value: TASKS_SEL, label: 'my fantastic label' });
    expect(props.getTasksSummary).toHaveBeenCalledWith({ EntityType: 'Arrears' });
  });

  it('should call the getArrearsSummary APIs with correct parameters when changing SORT filter', () => {
    el.setState({ currentResultsCount: 10 });
    el.instance().toggleArrearsSorting({ value: 'priority', label: 'my fantastic label' });
    expect(props.getArrearsSummary).toHaveBeenCalledWith(
      {
        arrearsStageId: null,
        offerType: null,
        sort: 'priority',
        status: null,
      },
      { value: 'priority', label: 'my fantastic label', count: 10 }
    );
  });

  it('should call the getArrearsSummary APIs with correct parameters when changing STAGE filter', () => {
    el.instance().toggleArrearsStage({ value: 'my value', label: 'my fantastic label', count: 5 });
    expect(props.getArrearsSummary).toHaveBeenCalledWith(
      {
        arrearsStageId: 'my value',
        offerType: undefined,
        sort: 'priority',
        status: null,
      },
      { value: 'my value', label: 'my fantastic label', count: 5 }
    );
  });

  it('should call the getArrearsSummary APIs with correct parameters when setting STAGE filter to "All Stages"', () => {
    el.instance().toggleArrearsStage({ value: '', label: 'All Stages' });
    expect(props.getArrearsSummary).toHaveBeenCalledWith(
      {
        arrearsStageId: '',
        offerType: undefined,
        sort: 'priority',
        status: null,
      },
      { value: '', label: 'All Stages' }
    );
  });

  it('should show the correct items in the stage filter when a status is selected', () => {
    el.setState({ statusFilterId: 1, statusFilterName: 'In Progress' });
    el.update();
    el.instance().toggleArrearsOrTasks({
      value: 857920000,
      label: 'In Progress',
      name: 'In Progress',
    });
    el.update();
    expect(el.state('stageItems')).toEqual([
      { label: 'All Stages', value: '' },
      { disabled: true, label: 'Leasehold', value: null },
      {
        count: 6,
        id: 'Stage 2',
        label: '    Stage 2 (6)',
        name: 'Stage 2',
        product: 'Leasehold',
        value: '4cb2fc58-4b75-e811-80c5-00155d072d63',
      },
      { disabled: true, label: 'PRH', value: null },
      {
        count: 2,
        id: '3wks Arrears',
        label: '    3wks Arrears (2)',
        name: '3wks Arrears',
        product: 'PRH',
        value: '4cb2fc58-4b75-e811-80c5-00155d072d63',
      },
      {
        count: 2,
        id: '< 3wks Arrears',
        label: '    < 3wks Arrears (2)',
        name: '< 3wks Arrears',
        product: 'PRH',
        value: '68027d4f-4b75-e811-80c5-00155d072d63',
      },
    ]);
  });

  it('should call getArrearStatuses on init', () => {
    expect(props.getArrearStatuses).toHaveBeenCalled();
  });
});
