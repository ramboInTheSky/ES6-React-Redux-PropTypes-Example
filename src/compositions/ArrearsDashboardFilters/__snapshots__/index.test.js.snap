// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArrearsFiltersComposition /> should render the page 1`] = `
<React.Fragment>
  <div
    className="row"
  >
    <styled.div
      data-bdd="ArrearsFilters-ArrearsFilters"
    >
      <Select
        autocomplete={false}
        dataBdd=""
        disabled={false}
        hasError={false}
        hideLabel={false}
        inline={true}
        inputValue="Arrears - All (11)"
        isFullWidth={true}
        itemToString={[Function]}
        items={
          Array [
            Object {
              "id": "Tasks",
              "label": "Tasks ",
              "value": "Tasks",
            },
            Object {
              "count": 11,
              "label": "Arrears - All (11)",
              "name": "All",
              "value": null,
            },
            Object {
              "count": 10,
              "label": "Arrears - In Progress (10)",
              "name": "In Progress",
              "value": 1,
            },
            Object {
              "count": 1,
              "label": "Arrears - Manually Paused (1)",
              "name": "Manually Paused",
              "value": 857920000,
            },
          ]
        }
        labelText="Show"
        onChange={[Function]}
        placeholder=""
      />
    </styled.div>
    <Select
      autocomplete={false}
      dataBdd=""
      disabled={false}
      hasError={false}
      hideLabel={false}
      inline={true}
      inputValue="Priority"
      isFullWidth={true}
      itemToString={[Function]}
      items={
        Array [
          Object {
            "id": "priority",
            "label": "Priority",
            "value": "priority",
          },
          Object {
            "id": "currentBalance",
            "label": "Current Balance",
            "value": "currentBalance",
          },
          Object {
            "id": "daysInArrears",
            "label": "Days in Arrears",
            "value": "daysInArrears",
          },
        ]
      }
      labelText="Sort By"
      onChange={[Function]}
      placeholder=""
    />
  </div>
  <div
    className="row"
  >
    <styled.div
      data-bdd="ArrearsFilters-StageFilter"
    >
      <Select
        autocomplete={false}
        dataBdd=""
        disabled={false}
        hasError={false}
        hideLabel={false}
        inline={true}
        inputValue=""
        isFullWidth={true}
        itemToString={[Function]}
        items={Array []}
        labelText="Stage"
        onChange={[Function]}
        placeholder="Select a Stage"
      />
    </styled.div>
  </div>
</React.Fragment>
`;
